llama-index-integrations/vector_stores/llama-index-vector-stores-databricks-vector-search/llama_index/vector_stores/databricks-vector-search/base.py

```diff
@@ -0,0 +1,399 @@
+        # Validate specified columns are in the index
+        index_schema = self._index_schema()
+
+        if self._is_direct_access_index() and index_schema:
+            missing_columns = columns_to_add - set(index_schema.keys())
+
+            if missing_columns:
+                raise ValueError(
+                    f"columns missing from schema: {', '.join(missing_columns)}"
+                )
+
+    def add(
+        self,
+        nodes: List[BaseNode],
+        **add_kwargs: Any,
+    ) -> List[str]:
+        """Add nodes to index.
+
+        Args:
+            nodes: List[BaseNode]: list of nodes with embeddings
+
+        """
+        if self._is_databricks_managed_embeddings():
+            raise ValueError(
+                "Adding nodes is not supported for Databricks-managed embeddings."
+            )
+
+        # construct the entries to upsert
+        entries = []
+        ids = []
+        for node in nodes:
+            node_id = node.node_id
+            metadata = node_to_metadata_dict(node, remove_text=True, flat_metadata=True)
+            entry = {
+                self._primary_key: node_id,
+                self.text_column: node.get_content(),
+                self._embedding_vector_column_name(): node.get_embedding(),
+                **{
+                    col: metadata.get(col)
+                    for col in filter(
+                        lambda column: column
+                        not in (self._primary_key, self.text_column),
+                        self.columns or [],
+                    )
+                },
+            }
+
+            entries.append(entry)
+            ids.append(node_id)
+
+        # attempt the upsert
+        upsert_resp = self._index.upsert(
+            entries,
+        )
+
+        # return the successful IDs
+        response_status = upsert_resp.get("status")
+
+        failed_ids = (
+            set(upsert_resp["result"]["failed_primary_keys"] or [])
+            if "result" in upsert_resp
+            and "failed_primary_keys" in upsert_resp["result"]
+            else set()
+        )
+
+        if response_status not in ("PARTIAL_SUCCESS", "FAILURE") or not failed_ids:
+            return ids
+
+        elif response_status == "PARTIAL_SUCCESS":
+            _logger.warning(
+                "failed to add %d out of %d texts to the index",
+                len(failed_ids),
+                len(ids),
+            )
+
+        elif response_status == "FAILURE":
+            _logger.error("failed to add all %d texts to the index", len(ids))
+
+        return list(filter(lambda id_: id_ not in failed_ids, ids))
+
+    def delete(self, ref_doc_id: str, **delete_kwargs: Any) -> None:
+        """
+        Delete nodes with ref_doc_id.
+
+        Args:
+            ref_doc_id (str): The doc_id of the document to delete.
+
+        """
+        self._index.delete(
+            primary_keys=[ref_doc_id],
```

Comment:
This isn't actually correct. The intention here is to delete any node with `doc_id == ref_doc_id` in its metadata. Is this possible here?